react-main/src/app/components/Game/index.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react';
+
+import Board from '../Board';
+
+import styles from './styles.scss';
+import calculateWinner from './utils.js';
+
+class Game extends Component {
+  state = { history: [{ squares: Array(9).fill(null)}], xIsNext: true, stepNumber: 0 };
+
+  jumpTo = move => {
+    this.setState({ stepNumber: move, xIsNext: move % 2 === 0 });
+  };
+
+  handleClick = i => {
+    const history = this.state.history;
+    const squares = this.currentHistory(history);
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([{ squares }]),
+      xIsNext: !this.state.xIsNext,
+      stepNumber: this.state.stepNumber+1
+    });
+  };
+
+  currentHistory = history => {
```

Comment:
You can do `currentHistory = history => history[this.state.stepNumber].squares.slice();`